storage/entities/block.js

```diff
@@ -14,8 +14,343 @@
+ * @property {string} [reward_eql]
+ * @property {string} [reward_ne]
+ * @property {string} [reward_gt]
+ * @property {string} [reward_gte]
+ * @property {string} [reward_lt]
+ * @property {string} [reward_lte]
+ * @property {string} [reward_in]
+ * @property {string} [version]
+ * @property {string} [version_eql]
+ * @property {string} [version_ne]
+ * @property {string} [version_gt]
+ * @property {string} [version_gte]
+ * @property {string} [version_lt]
+ * @property {string} [version_lte]
+ * @property {string} [version_in]
+ */
+
+/**
+ * @typedef {string} fieldSets.Block
+ * @enum
+ * @value 'FIELD_SET_SIMPLE'
+ */
+
+const FIELD_SET_SIMPLE = Symbol('FIELD_SET_SIMPLE');
+
+class Block extends BaseEntity {
+	/**
+	 * Constructor
+	 * @param {BaseAdapter} adapter - Adapter to retrive the data from
+	 * @param {filters.Block} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.overrideDefaultOptions({ fieldSet: FIELD_SET_SIMPLE });
+
+		this.addField('id', 'string', { filter: ft.TEXT });
+		this.addField('height', 'number', { filter: ft.NUMBER });
+		this.addField(
+			'blockSignature',
+			'string',
+			{ filter: ft.TEXT },
+			stringToByte
+		);
+		this.addField(
+			'generatorPublicKey',
+			'string',
+			{
+				format: 'publicKey',
+				filter: ft.TEXT,
+			},
+			stringToByte
+		);
+		this.addField('payloadHash', 'string', { filter: ft.TEXT }, stringToByte);
+		this.addField('payloadLength', 'number', { filter: ft.NUMBER });
+		this.addField('numberOfTransactions', 'number', { filter: ft.NUMBER });
+		this.addField('previousBlockId', 'string', {
+			filter: ft.TEXT,
+			fieldName: 'previousBlock',
+		});
+		this.addField('timestamp', 'number', { filter: ft.NUMBER });
+		this.addField('totalAmount', 'string', { filter: ft.NUMBER });
+		this.addField('totalFee', 'string', { filter: ft.NUMBER });
+		this.addField('reward', 'string', { filter: ft.NUMBER });
+		this.addField('version', 'number', { filter: ft.NUMBER });
+
+		this.SQLs = {
+			selectSimple: this.adapter.loadSQLFile('blocks/get_simple.sql'),
+			create: this.adapter.loadSQLFile('blocks/create.sql'),
+			isPersisted: this.adapter.loadSQLFile('blocks/is_persisted.sql'),
+		};
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	get FIELD_SET_SIMPLE() {
+		return FIELD_SET_SIMPLE;
+	}
+
+	/**
+	 * Get list of blocks
+	 *
+	 * @param {filters.Block|filters.Block[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {fieldSets.Block} [options.fieldSet='FIELD_SET_SIMPLE'] - Fieldset to choose
+	 * @param {Object} tx - Database transaction object
+	 * @return {Promise.<BasicBlock[], NonSupportedFilterTypeError|NonSupportedOptionError>}
+	 */
+	get(filters = {}, options = {}, tx) {
```

Comment:
Consider extracting the common part with `getOne` into a separate function.

Reply:
Will be discussed and added in another PR